ds4drv/__main__.py

```diff
@@ -108,26 +109,55 @@ def read_report(self):
     def run(self):
         self.loop.run()
 
-    def exit(self, *args):
+    def exit(self, *args, error = True):
         if self.device:
             self.cleanup_device()
 
-        self.logger.error(*args)
-        self.error = True
+        if error == True:
+            self.logger.error(*args)
+            self.error = True
+        else:
+            self.logger.info(*args)
 
 
 def create_controller_thread(index, controller_options, dynamic=False):
     controller = DS4Controller(index, controller_options, dynamic=dynamic)
 
     thread = Thread(target=controller.run)
-    thread.daemon = True
     thread.controller = controller
     thread.start()
 
     return thread
 
 
+class SigintHandler(object):
+
```

Comment:
I wouldn't say no need, that's my usual style. Doesn't fit the rest of the project though, so I'll clean it up.